tests/test_scalars.py

```diff
@@ -0,0 +1,234 @@
+    NaTType,
+)
+
+if TYPE_CHECKING:
+    from pandas.core.series import (
+        PeriodSeries,
+        TimedeltaSeries,
+    )
+
+    from pandas._typing import np_ndarray_bool
+else:
+    PeriodSeries = TimedeltaSeries = np_ndarray_bool = Any
+
+from tests import check
+
+from pandas.tseries.offsets import Day
+
+
+def test_period_construction() -> None:
+    p = pd.Period("2012-1-1", freq="D")
+    check(assert_type(p, pd.Period), pd.Period)
+    check(assert_type(pd.Period(p), pd.Period), pd.Period)
+    check(assert_type(pd.Period("2012-1-1", freq=Day()), pd.Period), pd.Period)
+    check(
+        assert_type(pd.Period(freq="D", year=2012, day=1, month=1), pd.Period),
+        pd.Period,
+    )
+    check(
+        assert_type(pd.Period(None, "D", year=2012, day=1, month=1), pd.Period),
+        pd.Period,
+    )
+    check(
+        assert_type(pd.Period(None, "D", 1, year=2012, day=1, month=1), pd.Period),
+        pd.Period,
+    )
+    check(
+        assert_type(
+            pd.Period(
+                freq="s", year=2012, month=1, day=1, hour=12, minute=30, second=45
+            ),
+            pd.Period,
+        ),
+        pd.Period,
+    )
+    check(assert_type(pd.Period(freq="Q", year=2012, quarter=2), pd.Period), pd.Period)
+
+
+def test_period_properties() -> None:
+    p = pd.Period("2012-1-1", freq="D")
+
+    check(assert_type(p.day, int), int)
+    check(assert_type(p.day_of_week, int), int)
+    check(assert_type(p.day_of_year, int), int)
+    check(assert_type(p.dayofweek, int), int)
+    check(assert_type(p.dayofyear, int), int)
+    check(assert_type(p.days_in_month, int), int)
+    check(assert_type(p.daysinmonth, int), int)
+    check(assert_type(p.end_time, pd.Timestamp), pd.Timestamp)
+    check(assert_type(p.freqstr, str), str)
+    check(assert_type(p.hour, int), int)
+    check(assert_type(p.is_leap_year, bool), bool)
+    check(assert_type(p.minute, int), int)
+    check(assert_type(p.month, int), int)
+    check(assert_type(p.quarter, int), int)
+    check(assert_type(p.qyear, int), int)
+    check(assert_type(p.second, int), int)
+    check(assert_type(p.start_time, pd.Timestamp), pd.Timestamp)
+    check(assert_type(p.week, int), int)
+    check(assert_type(p.weekday, int), int)
+    check(assert_type(p.weekofyear, int), int)
+    check(assert_type(p.year, int), int)
+    check(assert_type(p.freq, BaseOffset), Day)
+    check(assert_type(p.ordinal, int), int)
+
+    p2 = pd.Period("2012-1-1", freq="2D")
+    check(assert_type(p2.freq, BaseOffset), Day)
+
+
+def test_periof_add_subtract() -> None:
+    p = pd.Period("2012-1-1", freq="D")
+
+    as0 = pd.Timedelta(1, "D")
+    as1 = dt.timedelta(days=1)
+    as2 = np.timedelta64(1, "D")
+    as3 = np.int64(1)
+    as4 = int(1)
+    as5 = pd.period_range("2012-1-1", periods=10, freq="D")
+    as6 = pd.Period("2012-1-1", freq="D")
+    scale = 24 * 60 * 60 * 10**9
+    as7 = cast(TimedeltaSeries, pd.Series(pd.timedelta_range(scale, scale, freq="D")))
```

Comment:
This cast shouldn't be needed, because I thought if you pass a `TimeDeltaIndex` to `pd.Series`, you get a `TimdeltaSeries`, and if not, we should fix that.

Reply:
I've fixed this.